File: frontend-app/src/core/model/dto/response/UsuarioResponse.ts
import type { UsuarioPerfil } from "../types";
import { DEFAULT_EMPRESA_OBJ, type EmpresaResponse } from "./EmpresaResponse";
import type { UsuarioGrupoResponse } from "./UsuarioGrupoResponse";

export interface UsuarioResponse {
    id : number;
    nome : string;
    email : string;
    username : string;
    perfil: UsuarioPerfil;
    empresa : EmpresaResponse;
    grupos : UsuarioGrupoResponse[];
}

export const DEFAULT_USUARIO_OBJ : UsuarioResponse = {
    id: 0,
    nome: '',
    email: '',
    username: '',
    perfil: 'ADMIN',
    empresa: DEFAULT_EMPRESA_OBJ,
    grupos: []
};

File: main-api/src/main/java/com/redemonitor/main/dto/response/UsuarioResponse.java
package com.redemonitor.main.dto.response;

import java.util.List;

import com.redemonitor.main.enums.UsuarioPerfil;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class UsuarioResponse {

    private Long id;

    private String nome;

    private String email;

    private String username;

    private UsuarioPerfil perfil;
    
    private EmpresaResponse empresa;

    private List<UsuarioGrupoResponse> grupos;

}
